src/main/java/com/aws/greengrass/util/Exec.java

```diff
@@ -74,6 +74,7 @@
 
     protected final AtomicBoolean isClosed = new AtomicBoolean(false);
     protected Process process;
+    protected int pid;
```

Comment:
I'd highly recommend making this a method, not a field. `getPid`, then subclasses can handle that however they want. This way is forcing a particular implementation.